treeherder/webapp/api/performance_serializers.py

```diff
@@ -146,3 +150,49 @@ class IssueTrackerSerializer(serializers.ModelSerializer):
     class Meta:
         model = IssueTracker
         fields = ['id', 'text', 'issueTrackerUrl']
+
+
+class PerformanceQueryParamsSerializer(serializers.Serializer):
+    startday = serializers.DateTimeField(required=False, allow_null=True, default=None)
+    endday = serializers.DateTimeField(required=False, allow_null=True, default=None)
+    revision = serializers.CharField(required=False, allow_null=True, default=None)
+    repository = serializers.CharField()
+    framework = serializers.ListField(child=serializers.IntegerField())
+    interval = serializers.IntegerField(required=False, allow_null=True, default=None)
+    parent_signature = serializers.CharField(required=False, allow_null=True, default=None)
+    no_subtests = serializers.BooleanField(required=False)
+
+    def validate(self, data):
+        if data['revision'] is None and (data['startday'] is None or data['endday'] is None):
+            raise serializers.ValidationError('Required: revision or startday and endday.')
+
+        return data
+
+    def validate_repository(self, repository):
+        try:
+            Repository.objects.get(name=repository)
+
+        except ObjectDoesNotExist:
+            raise serializers.ValidationError('{} does not exist.'.format(repository))
+
+        return repository
+
+
+class PerformanceRevisionSerializer(serializers.ModelSerializer):
```

Comment:
This is a bit of a misnomer, as the API actually often returns results over *multiple* revisions. How about we call this `PerformanceSummarySerializer`? I'd also like to see a brief comment about exactly what this is and where it's used.

Reply:
Sure